nailgun-server/src/main/java/com/martiansoftware/nailgun/NGInputStream.java

```diff
@@ -40,15 +42,16 @@
     private final DataInputStream din;
     private InputStream stdin = null;
     private boolean eof = false;
-    private long remaining = 0;
+    private boolean clientConnected = true;
+    private int remaining = 0;
     private byte[] oneByteBuffer = null;
     private final DataOutputStream out;
     private boolean started = false;
-    private long lastReadTime = System.currentTimeMillis();
-    private final Future readFuture;
-    private final Set clientListeners = new HashSet();
-    private final Set heartbeatListeners = new HashSet();
+    private AtomicLong lastReadTime = new AtomicLong(System.currentTimeMillis());
+    private final Set<NGClientListener> clientListeners = new HashSet();
```

Comment:
hm, shouldn't this be something like `new HashSet<>()`? How come it's not a compiler warning?

Reply:
Maven configuration may be